sdk/identity/identity/src/credentials/managedIdentityCredential/utils.ts

```diff
@@ -73,3 +74,33 @@ export function parseExpirationTimestamp(body: TokenResponseParsedBody): number
     `Failed to parse token expiration from body. expires_in="${body.expires_in}", expires_on="${body.expires_on}"`,
   );
 }
+
+/**
+ * Given a token response, return the expiration timestamp as the number of milliseconds from the Unix epoch.
+ * @param body - A parsed response body from the authentication endpoint.
+ */
+export function parseRefreshTimestamp(body: TokenResponseParsedBody): number {
+  if (typeof body.refresh_on === "number") {
+    return body.refresh_on * 1000;
+  }
+
+  if (typeof body.refresh_on === "string") {
+    const asNumber = +body.refresh_on;
+    if (!isNaN(asNumber)) {
+      return asNumber * 1000;
+    }
+
+    const asDate = Date.parse(body.refresh_on);
+    if (!isNaN(asDate)) {
+      return asDate;
+    }
+  }
+
+  if (typeof body.refresh_in === "number") {
```

Comment:
is this correct? we're using refresh_in and refresh_on in the same function - but maybe its some legacy code I dont understand yet

Reply:
I just replicated the expiration date utility function for the refresh timestamp

Reply:
I am cleaning up the refresh_in because that was supposed to be removed as part of this PR https://github.com/Azure/azure-sdk-for-js/pull/25535 .See https://github.com/Azure/azure-sdk-for-js/pull/30402/files#r1712215938